### PR TITLE
Fix docker container permissions for ms2rescore

### PR DESCRIPTION
**Pull request by jonasscheid**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test_percolator.config [hunk 1]
```diff
@@ -25,5 +25,5 @@ params {
 
     // Don't do quantification since this step needs a larger test dataset (-> test quantification using test_full)
     skip_quantification = true
-    rescoring_engine    = 'mokapot'
+    rescoring_engine    = 'percolator'
```
Comment (marissaDubbelaar) [inline conf/test_percolator.config:28]: What is the reason that this is set back to percolator?

File: conf/modules.config [hunk 2]
```diff
@@ -157,6 +157,7 @@ process {
     }
 
     withName: 'MS2RESCORE' {
+        containerOptions = '-u $(id -u) -e "HOME=${HOME}" -v $HOME:$HOME'
```
Comment (marissaDubbelaar) [inline conf/modules.config:160]: Does this work like this or does it need to be `process.containerOptions`?

Comment (jonasscheid) [reply conf/modules.config:160]: `process.containerOptions` works only in the nextflow.config

I oriented myself by Julia's fix: https://github.com/nf-core/tools/pull/2673

By this we can define container options on module level, that is not possible with `runOptions`

Comment (jonasscheid) [reply conf/modules.config:160]: And: The MS2Rescore container needed a specific "treatment" in order to run, else permission issues. Previously that was handled by `userEmulation`, but that is/will be removed :/

Comment (jonasscheid) [reply conf/modules.config:160]: Ok so this works when having sudo rights, but apparently not via GH actions.. Im gonna test the full diff compared to userEmulation

**Comment by jonasscheid** [comment]
To fix lints we need to wait for the newest template update. Conversation: https://nfcore.slack.com/archives/CE6PELWR4/p1705931019545439?thread_ts=1705482353.145949&cid=CE6PELWR4

**Review by marissaDubbelaar — APPROVED** [review]
👍

**Pull request merged**